### PR TITLE
ISPN-14817 Move some Hot Rod client tests to the server testsuite

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.redhat.com/browse/ISPN-14817

-  ClusteredIT: incorporate ExtensionsIT suite run in CONTAINER mode
- HotRodCacheOperations: incorporate APITest and RemoteAsyncAPITest. Parameterized for protocol versions and types
- HotRodCacheQueries: incorporate ProtobufHotRodRestEntityIT
- HotRodCacheEvents: incorporate JsonEventsTest, ClientEventsTest, ClientFilterEventsTest, ClientCustomEventsTest, ClientClusterEventsTest
- SslTest, SslAuthenticationTest, SecureExecTest: removed as more in-depth tests are already present in the server testsuite
 

Part 1 of many. Do not close the Jira

### CODE REVIEW

**Comment by tristantarrant** [comment]
Failures are unrelated

File: quarkus/integration-tests/server/src/test/resources/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM registry.access.redhat.com/ubi8/ubi-minimal
+FROM registry.access.redhat.com/ubi9/ubi-minimal
```
Comment (ryanemerson) [inline quarkus/integration-tests/server/src/test/resources/Dockerfile:1]: We actually use ubi8 for the upstream images, as we needed to baseline the JDK based image on that for Openshift compatibility. That being said, I should probably upgrade the cli and server-native images to ubi9 as we don't have similar downstream considerations.

**Comment by ryanemerson** [comment]
Thanks @tristantarrant

File: quarkus/integration-tests/server/src/test/resources/Dockerfile (hunk 1 above)
Comment (tristantarrant) [reply quarkus/integration-tests/server/src/test/resources/Dockerfile:1]: Sorry, that's a leftover from running it on my local box which has a much newer glibc than CI.

**Pull request merged**